hooks/useGetUser.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+import useArConnect from 'use-arconnect';
+import useWalletAddress from './useWalletAddress';
+
+interface UserProps {
+  userName: string;
+  walletAddress: string;
+}
+
+const useGetUser = () => {
+  const [allUsers, setAllUsers] = useState<UserProps[]>([]),
+    [loading, setLoading] = useState(false),
+    address = useWalletAddress(),
+    arconnect = useArConnect();
+
+  let currentUser: UserProps = { userName: '', walletAddress: '' };
+
+  useEffect(() => {
+    if (arconnect) {
+      const getAllUsers = async () => {
```

Comment:
Don't declare functions inside possible loops like `useEffect`. This will create a scoped function everytime `useEffect` gets triggered. Really bad for memory and scoping